File: fireplace-app/src/main/java/com/github/bric3/fireplace/ui/HudPanel.java
package com.github.bric3.fireplace.ui;

import com.github.bric3.fireplace.DnDTarget;

import javax.swing.*;
import java.awt.*;

public class HudPanel {
    private final JPanel dndPanel;
    private final JPanel progressPanel;
    private final JPanel hudPanel;

    public HudPanel() {
        dndPanel = new GlassJPanel(new GridBagLayout());
        dndPanel.add(new JLabel("<html><font size=+4>Drag and drop JFR file here</font></html>"));
        dndPanel.setOpaque(false);
        dndPanel.setVisible(false);

        progressPanel = new GlassJPanel(new BorderLayout());
        var progress = new JProgressBar();
        progress.setIndeterminate(true);
        progressPanel.add(new JLabel("<html><font size=+4>Loading in progress</font></html>", SwingConstants.CENTER), BorderLayout.CENTER);
        progressPanel.add(progress, BorderLayout.SOUTH);

        hudPanel = new JPanel();
        hudPanel.setLayout(new BoxLayout(hudPanel, BoxLayout.Y_AXIS));
        hudPanel.add(dndPanel);
        hudPanel.add(progressPanel);
        hudPanel.setOpaque(false);
    }

    public JComponent getComponent() {
        return hudPanel;
    }

    public DnDTarget getDnDTarget() {
        return new DnDTarget() {
            @Override
            public JComponent getComponent() {
                return hudPanel;
            }

            @Override
            public void activate() {
                progressPanel.setVisible(false);
                dndPanel.setVisible(true);
            }

            @Override
            public void deactivate() {
                dndPanel.setVisible(false);
            }
        };
    }

    public void setProgressVisible(boolean visible) {
        if (visible) {
            dndPanel.setVisible(false);
        }
        progressPanel.setVisible(visible);
    }
}

File: fireplace-app/src/main/kotlin/io/github/bric3/fireplace/ui/HudPanel.kt
/*
 * Fireplace
 *
 * Copyright (c) 2021, Today - Brice Dutheil
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */
package io.github.bric3.fireplace.ui

import java.awt.BorderLayout
import java.awt.GridBagLayout
import javax.swing.*

class HudPanel {
    private val dndPanel: JPanel
    private val progressPanel: JPanel
    private val hudPanel: JPanel

    init {
        dndPanel = GlassJPanel(GridBagLayout()).apply {
            add(JLabel("<html><font size=+4>Drag and drop JFR file here</font></html>"))
            setOpaque(false)
            setVisible(false)
        }
        progressPanel = GlassJPanel(BorderLayout()).apply {
            add(
                JLabel("<html><font size=+4>Loading in progress</font></html>", SwingConstants.CENTER),
                BorderLayout.CENTER
            )
            val progress = JProgressBar().apply {
                isIndeterminate = true
            }
            add(progress, BorderLayout.SOUTH)
        }
        hudPanel = JPanel().apply {
            layout = BoxLayout(this, BoxLayout.Y_AXIS)
            add(dndPanel)
            add(progressPanel)
            isOpaque = false
        }
    }

    val component: JComponent
        get() = hudPanel

    val dnDTarget: DragAndDropTarget
        get() = object : DragAndDropTarget {
            override val component = hudPanel

            override fun activate() {
                progressPanel.isVisible = false
                dndPanel.isVisible = true
            }

            override fun deactivate() {
                dndPanel.isVisible = false
            }
        }

    fun setProgressVisible(visible: Boolean) {
        if (visible) {
            dndPanel.isVisible = false
        }
        progressPanel.isVisible = visible
    }
}
